classes/Group/Group.mjs

```diff
@@ -0,0 +1,117 @@
+import dbDriver from "../../database/driver.mjs"
+const database = new dbDriver("mongo")
+
+export default class Group {
+    constructor(groupId) {
+        this._id = groupId
+    }
+
+    async #loadFromDB() {
+        return database.getById(this.groupId, "Group")
+    }
+
+    async getMembers() {
+        return this.members ?? await this.#loadFromDB()
+    }
+
+    addMember(memberId, roles) {
+        if (this.members[memberId]) {
+            throw {
+                status: 400,
+                message: "Member already exists"
+            }
+        }
+        this.members[memberId] = true
+        this.updateMember(memberId, roles)
+    }
+
+    updateMember(memberId, roles) {
+        if (!this.members[memberId]) {
+            throw {
+                status: 404,
+                message: "Member not found"
+            }
+        }
+        if(!Array.isArray(roles)) {
+            if(typeof roles !== "string") {
+                throw {
+                    status: 400,
+                    message: "Invalid roles"
+                }
+            roles = roles.split(" ")
+            }
+        }
+        this.members[memberId].roles = roles
+    }
+
+    addMemberRoles(memberId, roles) {
+        if (!this.members[memberId]) {
+            throw {
+                status: 404,
+                message: "Member not found"
+            }
+        }
+        if(!Array.isArray(roles)) {
+            if(typeof roles !== "string") {
+                throw {
+                    status: 400,
+                    message: "Invalid roles"
+                }
+            roles = roles.split(" ")
+            }
+        }
+        this.members[memberId].roles = [...new Set([this.members[memberId].roles.concat(roles)])]        
+    }
+
+    removeMemberRoles(memberId, roles) {
+        if (!this.members[memberId]) {
+            throw {
+                status: 404,
+                message: "Member not found"
+            }
+        }
+        if(!Array.isArray(roles)) {
+            if(typeof roles !== "string") {
+                throw {
+                    status: 400,
+                    message: "Invalid roles"
+                }
+            roles = roles.split(" ")
+            }
+        }
+        this.members[memberId].roles = this.members[memberId].roles.filter(role => !roles.includes(role))
+    }
+
+    removeMember(memberId) {
+        delete this.members[memberId]
+    }
+
+    getByRole(role) {
+        return Object.keys(this.members).filter(memberId => this.members[memberId].roles.includes(role))
+    }
+
+    async save() {
+        return database.save(this,process.env.TPENGROUPS)
+    }
+
+    static async createNewGroup(creator, payload) {
+        const { customRoles, label, members } = payload
+        if (!ownerId) {
```

Comment:
`ownerId `is not defined. Did you mean to extract `ownerId `from the payload?

Reply:
This is missing some alignment. It should be creator. I'm fixing it